packages/web/components/Landing/LandingConnectButton.tsx

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable no-nested-ternary */
+import {
+  Box,
+  Button,
+  Icon,
+  Spinner,
+  Text,
+  Tooltip,
+  useBreakpointValue,
+  useToast,
+  VStack,
+} from '@metafam/ds';
+import { MetaLink } from 'components/Link';
+import { useWeb3 } from 'lib/hooks';
+import { useEffect, useState } from 'react';
+import { GoSignIn, GoSignOut } from 'react-icons/go';
+import { shortenAddress } from 'utils/stringHelpers';
+
+export const LandingConnectButton = ({ isIconStyle = false, ...props }) => {
+  const {
+    connect,
+    disconnect,
+    connected,
+    connecting,
+    address,
+    chainId,
+    provider,
+  } = useWeb3();
+  const [wrongNetwork, setWrongNetwork] = useState(false);
+  const spinnerSize = useBreakpointValue({ base: 'sm', '2xl': 'md' });
+  const toast = useToast();
+
+  useEffect(() => {
+    if (connected && address && provider) {
+      const networkName = provider.network.name;
+      const shortAddress = shortenAddress(address);
+      console.log('Connected to', networkName, 'with address', shortAddress);
+
+      if (chainId !== '0x89') {
+        setWrongNetwork(true);
+        toast({
+          title: `Wallet Connection`,
+          description: (
+            <VStack spacing={2} justifyItems="start">
+              <Text>
+                <Box as="strong" textTransform="capitalize">
+                  {networkName ?? ''} ({chainId})
+                </Box>{' '}
+                network is not supported. This area of the website requires{' '}
+                <strong>Polygon (id: 0x89)</strong> network. Please switch
+                networks.
+              </Text>
+              <Text>
+                You can visit{' '}
+                <MetaLink href="https://chainlist.org/chain/137" isExternal>
+                  chainlist.org
+                </MetaLink>{' '}
+                to add Polygon to your wallet.
+              </Text>
+            </VStack>
+          ),
+          status: 'warning',
+          isClosable: true,
+          duration: 5000,
+        });
+      } else {
+        setWrongNetwork(false);
+        toast({
+          title: 'Wallet Connection',
+          description: `Successfully connected to ${networkName} with ${shortAddress}. 🥳`,
+          status: 'success',
+          isClosable: true,
+          duration: 5000,
+        });
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I see a few places where you're using `eslint-disable-next-line react-hooks/exhaustive-deps`. In my experience you're doing something wrong if you need this rule disabled. 

In any case you should be able to add whatever state variables inside the hook into the dependency array and this warning should go away :)

Reply:
> I see a few places where you're using `eslint-disable-next-line react-hooks/exhaustive-deps`. In my experience you're doing something wrong if you need this rule disabled.
> 
> In any case you should be able to add whatever state variables inside the hook into the dependency array and this warning should go away :)

this is interesting. i must be missing some key knowledge of correct use of useEffect (probably hooks in general) as oftentimes, when I include all deps, i just get endless re-renders. if I am doing it wrong,I need to address that here and in myself asap. 🙏🏻